From 3a464001351e56bbf91cc0d7b27806fff2059cae Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 20 Oct 2020 07:02:10 +0300
Subject: [PATCH] [SW-22029] goya2: increase stack size to 2k

Currently in arcboot flow, it is observed that maximum stack arcboot
uses is of 1260 bytes which is overflowed by 236 bytes from stack size
of 1024 bytes (1k). The stack overflow is mainly seen in arcboot key
validation flow while synopsys SDK API's are called. This doesn't
cause any corruption issue as stack uses 236 bytes of heap area which
is unused. This patch increase stack size to 2k in order to avoid any
stack overflow.

Change-Id: I8e031c8c05083381cc23483a792509fd6ba5f5d6
---

diff --git a/goya2_arc/arcboot_rom.lds b/goya2_arc/arcboot_rom.lds
index 39ba169..d7f132c 100644
--- a/goya2_arc/arcboot_rom.lds
+++ b/goya2_arc/arcboot_rom.lds
@@ -89,7 +89,7 @@
 	.stack : {
 		. = ALIGN(32 / 8);
 		__stack = . ;
-		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 1k)) ;
+		. = . + (DEFINED(__STACK_SIZE) ? __STACK_SIZE : (DEFINED(__DEFAULT_STACK_SIZE) ? __DEFAULT_STACK_SIZE : 2k)) ;
 		__stack_top = . ;
 	} > REGION_STACK
 }
\ No newline at end of file
